jailer/src/lib.rs

```diff
@@ -214,122 +215,102 @@ impl fmt::Display for Error {
 
 pub type Result<T> = result::Result<T, Error>;
 
-pub fn clap_app<'a, 'b>() -> App<'a, 'b> {
-    // Initially, the uid and gid params had default values, but it turns out that it's quite
-    // easy to shoot yourself in the foot by not setting proper permissions when preparing the
-    // contents of the jail, so I think their values should be provided explicitly.
-    App::new("jailer")
-        .version(crate_version!())
-        .author(crate_authors!())
-        .about("Jail a microVM.")
-        .setting(AppSettings::TrailingVarArg)
+fn sanitize_process() {
+    // First thing to do is make sure we don't keep any inherited FDs
+    // other that IN, OUT and ERR.
+    if let Ok(paths) = fs::read_dir("/proc/self/fd") {
+        for maybe_path in paths {
+            if maybe_path.is_err() {
+                continue;
+            }
+
+            let file_name = maybe_path.unwrap().file_name();
+            let fd_str = file_name.to_str().unwrap_or("0");
+            let fd = fd_str.parse::<i32>().unwrap_or(0);
+
+            if fd > 2 {
+                // Safe because close() cannot fail when passed a valid parameter.
+                unsafe { libc::close(fd) };
+            }
+        }
+    }
+}
+
+pub fn set_args_info(arg_parser: ArgParser) -> ArgParser {
```

Comment:
I wouldn't return an `ArgParser` here. It makes sense to do that as part of builder pattern when the method is defined inside the class. But using it here looks strange to me.